lib/chainer/serializers/marshal.rb

```diff
@@ -4,6 +4,12 @@ class MarshalSerializer < Chainer::Serializer
       attr_accessor :target, :path
 
       def self.save_file(filename, obj)
+        if filename.instance_of?(String)
+          File.open(filename, 'wb') do |f|
+            self.save_file(f, obj)
+          end
+          return
+        end
```

Comment:
I thought it strange that original implementation.
`File` object is passed to `filename`.


What do you think is natural thought as follows?

```
s = self.new
s.save(obj)
File.open(filename, 'wb') do |f|
   Marshal.dump(s.target, f)
end
```

Reply:
Well...
When calling this save_file function from elsewhere,  snapshot for example, the file object will be passed.

Reply:
I am planning to respond like this.
https://github.com/red-data-tools/red-chainer/pull/47/files

Reply:
I’ll leave it up to you.